File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        result = []
        for i,a in enumerate(nums):

            if(a>0): 
                break
            
            if i> 0 and a == nums[i-1]:
                continue

            j = i + 1
            k = len(nums) - 1

            while(j<k):
                if(-nums[i] == nums[j] + nums[k]): 
                    result.append([nums[i],nums[j],nums[k]])
                    j+=1
                    k-=1
                    while nums[j] == nums[j- 1] and j < k:
                        j += 1
                elif nums[j] + nums[k] > -nums[i]:
                    k-=1
                else:
                    j+=1

        return result
            
            



    
                    
           



      
                

        
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
     //napravi mapu: frequency->number
    map<int,int> frequencyMap;
    for(int num: nums){
        frequencyMap[num]++;
    }

    vector<pair<int,int>> arr;

    for( const auto& val:frequencyMap)

    arr.push_back({val.second, val.first});
    sort(arr.rbegin(), arr.rend());

    vector<int> result;
    for(int i=0;i<k;i++){
        result.push_back(arr[i].second);
    }
    return result;
    }
    
};
